flink-table/flink-table-planner/src/test/scala/org/apache/flink/table/planner/runtime/stream/sql/JoinITCase.scala

```diff
@@ -99,6 +106,33 @@ class JoinITCase(miniBatch: MiniBatchMode, state: StateBackendMode)
   // Tests for inner join.
   override def after(): Unit = {}
 
+  @TestTemplate
+  def test(): Unit = {
+    val tableA = failingDataSource(
+      Seq[(Int, Long, String)]((1, 1L, "Hi"), (2, 2L, "Hello"), (3, 3L, "Hello world")))
+      .toTable(tEnv, 'a1, 'a2, 'a3)
+    val tableB = failingDataSource(
+      Seq[(Int, Long, String)]((1, 1L, "Hi"), (2, 2L, "Hello"), (3, 3L, "Hello world")))
+      .toTable(tEnv, 'b1, 'b2, 'b3)
+    tEnv.createTemporaryView("C", tableA)
+    tEnv.createTemporaryView("D", tableB)
+
+    val sqlQuery = "SELECT * FROM C, D WHERE a1 = b1 and a2 = 1"
+
+    val sink = new TestingRetractSink
+    tEnv.sqlQuery(sqlQuery).toRetractStream[Row].addSink(sink).setParallelism(1)
+    env.execute()
+
+    val expected = mutable.Seq(
```

Comment:
Could you please clarify the rationale behind the expected sequence? IMO, Is it intended to be List(1, 1, 'Hi', 1, 1, 'Hi')?